File: app/src/debug/java/com/soumya/wwdablu/zomatobuddy/ZomatoBuddyApplication.java
package com.soumya.wwdablu.zomatobuddy;

import android.app.Application;

import com.soumya.wwdablu.zomatobuddy.common.Analytics;
import com.soumya.wwdablu.zomatobuddy.database.CacheDB;
import com.squareup.leakcanary.LeakCanary;

import io.realm.Realm;
import timber.log.Timber;

public class ZomatoBuddyApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();

        if (LeakCanary.isInAnalyzerProcess(this)) {
            return;
        }
        LeakCanary.install(this);

        //Init the CacheDB
        CacheDB.init(this);

        //Init analytics
        Analytics.init(this);

        //Realm init
        Realm.init(this);

        //Init debug version of Timber tree
        Timber.plant(new BuddyDebugTree());
    }
}

File: app/src/debug/java/com/soumya/wwdablu/zomatobuddy/ZomatoBuddyApplication.kt
package com.soumya.wwdablu.zomatobuddy

import android.app.Application
import timber.log.Timber

class ZomatoBuddyApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        //Init debug version of Timber tree
        Timber.plant(BuddyDebugTree())
    }
}